source/test/F0.Minesweeper.Logic.Tests/MinefieldTest.cs

```diff
@@ -17,5 +18,189 @@ public void MinefieldUncover_Should_ReturnAtLeastTheClickedLocation()
 			result.Cells.Should().NotBeEmpty()
 				.And.ContainSingle(cell => cell.Location == locationTestValue);
 		}
+
+		// 2x1 mine uncovered + Game Over
+		[Fact]
+		public void Uncover_OnCellWithMine_ReturnsMineAndGameLost()
+		{
+			var mineLocation = new Location(0, 0);
+			var minefieldUnderTest = new Minefield(2, 1, 1, new TestMinelayer(new List<Location> { mineLocation } ));
+
+			var result = minefieldUnderTest.Uncover(new Location(0,0));
+
+			result.Cells.Should().NotBeEmpty()
+				.And.ContainSingle(cell => cell.IsMine);
+			result.Status.Should().Be(GameStatus.IsLost);
+		}
+
+		// 2x1 empty uncovered + Win
+		[Fact]
+		public void Uncover_OnLastCellWithoutMine_ReturnsNoMineAndGameWon()
+		{
+			var mineLocation = new Location(1, 0);
+			var minefieldUnderTest = new Minefield(2, 1, 1, new TestMinelayer(new List<Location> { mineLocation }));
+
+			var result = minefieldUnderTest.Uncover(new Location(0, 0));
+
+			result.Cells.Should().NotBeEmpty()
+				.And.ContainSingle(cell => !cell.IsMine);
+			result.Status.Should().Be(GameStatus.IsWon);
+		}
+
+		// 3x1 Game still in progress
+		[Fact]
+		public void Uncover_OnCellWithoutMine_GameStillInProgress()
+		{
+			var mineLocation = new Location(1, 0);
+			var minefieldUnderTest = new Minefield(3, 1, 1, new TestMinelayer(new List<Location> { mineLocation }));
+
+			var result = minefieldUnderTest.Uncover(new Location(0, 0));
+
+			result.Cells.Should().NotBeEmpty()
+				.And.Contain(cell => !cell.IsMine);
+			result.Status.Should().Be(GameStatus.InProgress);
+		}
+
+		//3x3 all numbers
+		[Theory]
+		[MemberData(nameof(MinefieldTestData.TestData), MemberType = typeof(MinefieldTestData))]
+		public void Uncover_OnNoMine_ReturnsRightAmountOfAdjacentMines(MinefieldTestData testdata)
+		{
+			var minefieldUnderTest = new Minefield(3, 3, 0, new TestMinelayer(testdata.MineLocations));
+
+			var result = minefieldUnderTest.Uncover(new Location(1, 1));
+
+			result.Cells.Should().NotBeEmpty()
+				.And.ContainSingle(cell => cell.AdjacentMineCount == testdata.MineLocations.Count);
+		}
+
+		//5x5 Empty cell propagation + win
+		// [C][ ][ ][ ][ ]
+		// [ ][ ][ ][ ][ ]
+		// [M][M][M][ ][ ]
+		// [ ][ ][ ][ ][ ]
+		// [ ][ ][ ][ ][ ]
+		[Fact]
+		public void Uncover_OnEmptyArea_PropagatedThroughWholeMinefield()
+		{
+			var mineLocations = new List<Location> { new Location(0, 2), new Location(1, 2), new Location(2, 2) };
```

Comment:
Yep, also as we discussed, I will change all of these to _Target-typed new expressions_ 😄 
No more `var` for declaring new variables!! 😉

Reply:
One thing I noticed was that _Target-typed new expressions_ don't work for arrays? Or at least with array initialization? 🤔 
`List<Location> mineLocations = new(){ new(0, 3), new(1, 3), new(1, 4) };` This works
`Location[] mineLocations = new(){ new(0, 3), new(1, 3), new(1, 4) };` This doesn't.
I tried a few other variations but couldn't do it 🙈 
But maybe I'm overlooking something?

Reply:
Have you tried to initialize the array with square instead of round brackets? Because I believe this should work:
`Location[] mineLocations = new [] { new(0, 3), new(1, 3), new(1, 4) };`

Reply:
Thank you @dev-in-disguise for the correct solution/syntax for this puzzle 😄 
`Location[] mineLocations = { new(0, 3), new(1, 3), new(1, 4) };`